samza-core/src/main/java/org/apache/samza/container/LocalityManager.java

```diff
@@ -19,38 +19,75 @@
 
 package org.apache.samza.container;
 
-import org.apache.samza.container.grouper.task.TaskAssignmentManager;
-import org.apache.samza.coordinator.stream.CoordinatorStreamManager;
-import org.apache.samza.coordinator.stream.messages.CoordinatorStreamMessage;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import com.google.common.collect.ImmutableMap;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.JobConfig;
+import org.apache.samza.container.grouper.task.TaskAssignmentManager;
+import org.apache.samza.coordinator.stream.CoordinatorStreamKeySerde;
+import org.apache.samza.coordinator.stream.CoordinatorStreamValueSerde;
 import org.apache.samza.coordinator.stream.messages.SetContainerHostMapping;
-
+import org.apache.samza.metadatastore.MetadataStore;
+import org.apache.samza.metadatastore.MetadataStoreFactory;
+import org.apache.samza.metrics.MetricsRegistry;
+import org.apache.samza.serializers.Serde;
+import org.apache.samza.util.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Locality Manager is used to persist and read the container-to-host
- * assignment information from the coordinator stream
+ * assignment information from the coordinator stream.
  * */
 public class LocalityManager {
```

Comment:
This seems to be a wrapper over MetadataStore, specialized to store host-container mapping, so 
could we name it something specific like ContainerHostMappingStore? 
Or do you foresee extensions of this